### PR TITLE
Make credit computation configurable

### PR DESCRIPTION
**Pull request by Neverlord**
This PR makes the algorithm used for computing credit configurable. I've also added a "size-based" policy that estimates the size (in Bytes) of incoming streaming elements and constrains credit by an upper bound (64kb by default). I could improve performance in Broker with the new policy in a distributed setting, but the `simple_streaming` benchmark runs a bit slower. So, before we drop the current implementation (now called complexity-based controller) we'll have to do some more evaluation.

### CODE REVIEW

File: libcaf_core/src/size_based_credit_controller.cpp [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+void impl::after_processing(downstream_msg::batch&) {
+  // nop
+}
+
+credit_controller::assignment impl::compute_initial() {
+  return {buffer_size_, batch_size_};
+}
+
+credit_controller::assignment impl::compute(timespan, int32_t) {
+  if (sampled_elements_ > 9) {
```
Comment (josephnoir) [inline libcaf_core/src/size_based_credit_controller.cpp]: Why 9, is this related to the 10 batches per cycle below?

File: libcaf_core/src/complexity_based_credit_controller.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  num_elements_ += x.xs_size;
+  processing_begin_ = make_timestamp();
+}
+
+void impl::after_processing(downstream_msg::batch&) {
+  processing_time_ += make_timestamp() - processing_begin_;
+}
+
+credit_controller::assignment impl::compute_initial() {
+  return {50, 10};
```
[the remark this answers is not in the document]
Comment (Neverlord) [reply libcaf_core/src/complexity_based_credit_controller.cpp]: 🧙‍♂️
